ksql-streams/src/main/java/io/confluent/ksql/execution/streams/StreamSourceBuilder.java

```diff
@@ -40,7 +41,19 @@
   private StreamSourceBuilder() {
   }
 
-  public static KStream<Struct, GenericRow> buildUnwindowed(
+  @SuppressWarnings("unchecked")
+  public static <K> StreamWithKeySerdeFactory<K> build(
+      final KsqlQueryBuilder queryBuilder,
+      final StreamSource<?> streamSource
+  ) {
+    if (streamSource.getFormats().getKeyFormat().isWindowed()) {
+      return (StreamWithKeySerdeFactory) buildWindowed(queryBuilder,streamSource);
```

Comment:
This is not ideal. Planning to clean this up in a follow-on that adds a separate step type for a windowed source so we can have it strictly typed (similar to aggregates).

Reply:
+1